### PR TITLE
Register Provider Extrinsic

### LINKED ISSUE #319: [Provider Registration] Register as a Provider
As a service, I want to be able to register my MSA as a Provider so that others can see that I am one and still be able to be delegated to once the limit is imposed (#320).

## Design Doc
- https://github.com/LibertyDSNP/frequency/blob/main/designdocs/provider_registration.md

## Acceptance Criteria
- [x] Assume that an MSA with one or more keys already exists.
- [x] I should be able to call an extrinsic on the MSA pallet that allows me to register my MSA as a provider
  - [x] I should be able to set a utf-8 name for my provider (*non-unique*)
- [x] Registration of a provider MSA must be from a key associated with the MSA
- [ ] Limit the name to 16 bytes (4-16 UTF-8 Chars)
- [x] Benchmarks and weights

## Arch Notes
- [Extrinsic definition in Design Doc](https://github.com/LibertyDSNP/frequency/blob/main/designdocs/provider_registration.md#register_providerorigin-registration_params)
- This is a token transaction
- Storage
  - key: msa_id
  - Value: ProviderMetadata { name }
- Code should be added into the msa_pallet
- Errors
  - Cannot register the same provider twice Naming should take https://github.com/LibertyDSNP/frequency/pull/308/files#r942482857 into account
  - Origin not associated to an MSA
  - Metadata Parameter out of bounds

## Related Stories
- #321 
- #322 
- #320 
- #344

### PR DESCRIPTION
**Pull request by dopatraman**
# Goal
The goal of this PR is to introduce an extrinsic for registering a provider.

Closes #319 

# Discussion
This PR covers adding a new Provider to storage and associating it with an MSA.

# Checklist
- [x] Updated the js/api-augment code if a custom RPC added/changed
- [x] Design doc(s) updated
- [x] Tests added
- [x] Benchmarks added
- [x] Weights updated

### CODE REVIEW

**Review by enddynayn — APPROVED** [review]
Fantastic! 👍

File: pallets/msa/src/lib.rs [hunk 1]
```diff
@@ -297,6 +321,31 @@ pub mod pallet {
...
 		}
 
+		/// Adds an association between MSA id and ProviderMetadata. As of now, the
+		/// only piece of metadata we are recording is provider name.
+		///
+		/// ## Errors
+		/// - Returns
+		///   [`DuplicateProviderMetadata`](Error::DuplicateProviderMetadata) if there is already a ProviderMetadata associated with the given MSA id.
+		#[pallet::weight(T::WeightInfo::register_provider())]
+		pub fn register_provider(origin: OriginFor<T>, provider_name: Vec<u8>) -> DispatchResult {
```
Comment (wilwade) [inline pallets/msa/src/lib.rs:331]: Don't forget to add this to the api-augment code

Comment (dopatraman) [reply pallets/msa/src/lib.rs:331]: I updated `lookup.ts`. Are there any other files I should be concerned about updating?

File: pallets/msa/src/tests.rs [hunk 2]
```diff
@@ -1178,5 +1178,18 @@ fn add_removed_key_to_msa_pass() {
 			signature,
 			add_new_key_data
 		));
+	});
+}
+
+#[test]
+fn register_provider() {
```
Comment (shannonwells) [inline pallets/msa/src/tests.rs:1185]: This needs negative tests for the error conditions.

**Review by shannonwells — CHANGES_REQUESTED** [review]
I'd like to see tests for the error conditions; if nothing else it's a guard against regression/accidental removal

**Review by wilwade — APPROVED** [review]
Looks good on my side.

**Pull request merged**